modules/coredns/collect.go

```diff
@@ -61,14 +84,49 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 	return stm.ToMap(mx), nil
 }
 
-// Summary
+func (cd *CoreDNS) updateVersionDependentMetrics(raw prometheus.Metrics) {
+	version := cd.parseVersion(raw)
+	if version == nil {
+		return
+	}
```

Comment:
Should we version = version169 if we can't find a build info metric?

Reply:
I see **all** "FindByName" functions use "cd.metricNames" and we set the structure fields only if we manage to find/parse the version metric.

I think we always get "coredns_build_info" metric, but if not there are 2 ways

- if we don't try to guess, we should return immidiatly after https://github.com/georgeok/go.d.plugin/blob/5ccfd0b2f30319d5f92076953cac9829528ee54c/modules/coredns/collect.go#L57-L60 if cd.version == nil.
- or assume that version is X and try to process metrics.

I am confused because you added cd.version != check [here](https://github.com/georgeok/go.d.plugin/blob/5ccfd0b2f30319d5f92076953cac9829528ee54c/modules/coredns/collect.go#L62-L68), but not [there](https://github.com/georgeok/go.d.plugin/blob/5ccfd0b2f30319d5f92076953cac9829528ee54c/modules/coredns/collect.go#L70-L81).